ruleset.go

```diff
@@ -21,11 +21,11 @@ type Rule struct {
 // Ruleset is a set of rules to apply to a set of targets with flags for things
 // like whether or not the set is active, targets, rules, exclusions, etc.
 type Ruleset struct {
-	Off       string      `xml:"default_off,attr"`
-	Platform  string      `xml:"platform,attr"`
-	Target    []Target    `xml:"target"`
-	Exclusion []Exclusion `xml:"exclusion"`
-	Rule      []Rule      `xml:"rule"`
+	Off       string       `xml:"default_off,attr"`
+	Platform  string       `xml:"platform,attr"`
+	Target    []*Target    `xml:"target"`
+	Exclusion []*Exclusion `xml:"exclusion"`
+	Rule      []*Rule      `xml:"rule"`
```

Comment:
Made these all pointers to they can be modified without making copies and re-inserting into the arrays.